src/libply/arch/mips.c

```diff
@@ -0,0 +1,163 @@
+#ifndef USE_MIPS_64BIT
+	/* pad for 32-bit system; refer to linux/arch/mips/include/asm/ptrace.h */
+	{ .name = "uarg0",        .type = &t_reg_t },
+	{ .name = "uarg1",        .type = &t_reg_t },
+	{ .name = "uarg2",        .type = &t_reg_t },
+	{ .name = "uarg3",        .type = &t_reg_t },
+	{ .name = "uarg4",        .type = &t_reg_t },
+	{ .name = "uarg5",        .type = &t_reg_t },
+	{ .name = "uarg6",        .type = &t_reg_t },
+	{ .name = "uarg7",        .type = &t_reg_t },
+#endif
+
+	{ .name = "zero",         .type = &t_reg_t },    /* $0  */
+	{ .name = "at",           .type = &t_reg_t },    /* $1  */
+	{ .name = "v0",           .type = &t_reg_t },    /* $2  */
+	{ .name = "v1",           .type = &t_reg_t },    /* $3  */
+	{ .name = "a0",           .type = &t_reg_t },    /* $4  */
+	{ .name = "a1",           .type = &t_reg_t },    /* $5  */
+	{ .name = "a2",           .type = &t_reg_t },    /* $6  */
+	{ .name = "a3",           .type = &t_reg_t },    /* $7  */
+#ifndef USE_MIPS_N32 /* o32 ABI */
+	{ .name = "t0",           .type = &t_reg_t },    /* $8  */
+	{ .name = "t1",           .type = &t_reg_t },    /* $9  */
+	{ .name = "t2",           .type = &t_reg_t },    /* $10 */
+	{ .name = "t3",           .type = &t_reg_t },    /* $11 */
+#else /* n32 or native 64-bit ABI */
+	{ .name = "a4",           .type = &t_reg_t },    /* $8  */
+	{ .name = "a5",           .type = &t_reg_t },    /* $9  */
+	{ .name = "a6",           .type = &t_reg_t },    /* $10 */
+	{ .name = "a7",           .type = &t_reg_t },    /* $11 */
+#endif
+	{ .name = "t4",           .type = &t_reg_t },    /* $12 */
+	{ .name = "t5",           .type = &t_reg_t },    /* $13 */
+	{ .name = "t6",           .type = &t_reg_t },    /* $14 */
+	{ .name = "t7",           .type = &t_reg_t },    /* $15 */
+	{ .name = "s0",           .type = &t_reg_t },    /* $16 */
+	{ .name = "s1",           .type = &t_reg_t },    /* $17 */
+	{ .name = "s2",           .type = &t_reg_t },    /* $18 */
+	{ .name = "s3",           .type = &t_reg_t },    /* $19 */
+	{ .name = "s4",           .type = &t_reg_t },    /* $20 */
+	{ .name = "s5",           .type = &t_reg_t },    /* $21 */
+	{ .name = "s6",           .type = &t_reg_t },    /* $22 */
+	{ .name = "s7",           .type = &t_reg_t },    /* $23 */
+	{ .name = "t8",           .type = &t_reg_t },    /* $24 */
+	{ .name = "t9",           .type = &t_reg_t },    /* $25 */
+	{ .name = "k0",           .type = &t_reg_t },    /* $26 */
+	{ .name = "k1",           .type = &t_reg_t },    /* $27 */
+	{ .name = "gp",           .type = &t_reg_t },    /* $28 */
+	{ .name = "sp",           .type = &t_reg_t },    /* $29 */
+	{ .name = "fp",           .type = &t_reg_t },    /* $30, or s8 */
+	{ .name = "ra",           .type = &t_reg_t },    /* $31 */
+
+	{ .name = "cp0_status",   .type = &t_reg_t },
+	{ .name = "hi",           .type = &t_reg_t },
+	{ .name = "lo",           .type = &t_reg_t },
+#if 0 /* Let's pretend that CPU_HAS_SMARTMIPS could never be defined: */
```

Comment:
I'm not a MIPS guru. Why are we allowed to pretend that?

Reply:
On my MIPS32 Linux kernel build, `CONFIG_CPU_HAS_SMARTMIPS` is not defined; On buildroot/qemu_mips64 kernel build, the option is not defined either. I think the option is so rarely enabled, I would like to drop it; otherwise, another `--enable-smartmips` configuration option will be needed.